api/src/test/java/org/pmiops/workbench/reporting/ReportingUploadServiceTest.java

```diff
@@ -157,17 +197,54 @@ public void testUploadSnapshot_batchInserts() {
                 .fakeSize(4444L)
                 .creatorId(101L)));
 
-    reportingUploadService.uploadSnapshot(largeSnapshot);
+    reportingUploadServiceDmlImpl.uploadSnapshot(largeSnapshot);
     verify(mockBigQueryService, times(6))
         .executeQuery(queryJobConfigurationCaptor.capture(), anyLong());
 
     final List<QueryJobConfiguration> jobs = queryJobConfigurationCaptor.getAllValues();
     assertThat(jobs).hasSize(6);
-    final int researcherColumnCount = 4;
-    final int workspaceColumnCount = 5;
 
-    assertThat(jobs.get(0).getNamedParameters()).hasSize(researcherColumnCount * 5 + 1);
-    assertThat(jobs.get(4).getNamedParameters()).hasSize(researcherColumnCount + 1);
-    assertThat(jobs.get(5).getNamedParameters()).hasSize(workspaceColumnCount + 1);
+    assertThat(jobs.get(0).getNamedParameters()).hasSize(RESEARCHER_COLUMN_COUNT * 5 + 1);
+    assertThat(jobs.get(4).getNamedParameters()).hasSize(RESEARCHER_COLUMN_COUNT + 1);
+    assertThat(jobs.get(5).getNamedParameters()).hasSize(WORKSPACE_COLUMN_COUNT + 1);
+
+    final QueryParameterValue creationTime =
+        jobs.get(5).getNamedParameters().get("creation_time__0");
+    assertThat(creationTime).isNotNull();
+    final Instant instant = QueryParameterValues.timestampQpvToInstant(creationTime);
+    assertThat((double) instant.toEpochMilli()).isWithin(500.0).of(THEN.toEpochMilli());
+  }
+
+  @Test
+  public void testUploadSnapshot_streaming() {
```

Comment:
@jmthibault79 here is the streaming test. It's actually a bit harder to unit test, because not all of the conversions are documented (which gave me fits).